spacemacs-core: Fix issues with major-mode leader

justbur opened the pull request:
`mode-map` and `major-mode-map` were not let bound and had global scope. Seems like a bad idea for variables named like this

justbur commented:
Added another change related to one of the issues brought up in #3000. It makes sense to use the same hook that evil-leader-mode does in binding the major-mode-leader key, so that we don't get the inconsistency between the leader being bound without the major-mode leader key being bound, which seems to be a result of one hook successfully firing and the previously used hook for the major-mode leader key not firing (because of an error in setting up the major-mode).

TheBB commented:
_YES_, that's a good solution. :100:

TheBB commented:
Although I should like to have it in both hooks, so that it updates if one actually *does* change the major mode.

justbur commented:
@TheBB It will as it is here. evil-leader has to worry about the major-mode changing too. I'm just following it

TheBB commented:
Hm, you're right, it works.

justbur commented:
The idea is that it's as if we just rewrote evil-leader and added the code to the minor-mode initialization. For spacemacs we essentially need a version of evil-leader that support multiple leader keys

syl20bnr commented:
Having our own evil-leader implementation can be considered. Also it may be useful to support minor modes if it is possible.

Thank you :+1: 
I moved the functions to a new file `core-keybindings.el`. 

@TheBB @justbur I just realized that the choice of `spacemacs-core` for the base layer is not very smart... we already have a core in spacemacs... Do you have some alternative proposition while we can change it ?

Cherry-picked into develop branch, you can safely delete your branch.

TheBB commented:
Kernel? Essentials?

TheBB commented:
http://i.imgur.com/i7LP5V9.png

justbur commented:
I like base layer. Something to build on.
On Mon, Sep 14, 2015 at 1:14 AM Eivind Fonn <notifications@github.com>
wrote:

> http://i.imgur.com/i7LP5V9.png
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/pull/3002#issuecomment-139962609>.
>

syl20bnr commented:
`spacemacs-base` ? I would prefer namespacing it so people are free to define a base layer for their own use.

justbur commented:
Oh yes, sorry. That's what I meant

On Tue, Sep 15, 2015 at 9:29 AM, Sylvain Benner <notifications@github.com>
wrote:

> spacemacs-base ? I would prefer namespacing it so people are free to
> define a base layer for their own use.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/pull/3002#issuecomment-140393516>.
>

robbyoconnor commented:
I renamed `spacemacs-core` to `spacemacs-base` in #3048
